### ISSUE TITLE
In the PTK terminal, control-o should indent the newly created line

### ISSUE THREAD

**Issue by njsmith** [ipython/ipython#9588]
Suppose I have:

```
In [5]: def f():
   ...:     [CURSOR]a = 1           
```

If I press control-O to open up a new line, I get:

```
In [5]: def f():
   ...:     [CURSOR]
   ...: a = 1
```

I would prefer to get:

```
In [5]: def f():
   ...:     [CURSOR]
   ...:     a = 1
```

**Comment by Carreau** [comment]
I'm going to tag that as "new feature", that would be one more carrot for Python 3 once implemented :-P. 
I might have a got at that before but not promises !

**Comment by gforsyth** [comment]
@njsmith -- what should happen if you have 
```console
In [5]: def f():
     ...:     a = 1[CURSOR]
```

and hit Ctrl-O.  Do you expect a dedent or a preserved indentation level?

**Comment by njsmith** [comment]
I guess it doesn't matter that much? But an indent seems slightly more convenient?

I guess the general proposal would be: hitting control-O should be equivalent to hitting Enter, except that: (a) it should never cause the current block to be executed, (b) it should leave the cursor where it started.

**Comment by Carreau** [comment]
THis has been partially addressed by #9770 (well, all things listed in here seem to work). 
The one which is weird is : 

```
def foo():[CURSOR]
    a = 1$
````
`Ctrl-O` end up with:

```
def foo():[CURSOR]
$
    a = 1$
```
Which is still copying indent (if it's a method you are still at class level at least).
But we appear to have the logic for dedenting one level after `pass`, `return` and indenting one more after `:` so we should be able to improve.
